common/lib/controller/sub.controller.js

```diff
@@ -63,8 +64,46 @@ define(function(require, exports, module) {
 
   SubController.prototype.parseViewName = parseViewName;
 
-  SubController.prototype.handlePortMessage = function(port) {
-    throw new Error('Abstract method.');
+  /**
+   * Can be overriden to use the old event handling style. Alternatively
+   * the new 'ctrl.on('event', function(options) {})' style can be used.
+   * @param  {String} options.event   The event descriptor
+   * @param  {Object} options         Contains message attributes and data
+   */
+  SubController.prototype.handlePortMessage = function(options) {
+    options = options || {};
+    if (this._handlers.has(options.event)) {
+      this._handlers.get(options.event).call(this, options);
+    } else {
+      console.log('Unknown event', options);
+    }
+  };
```

Comment:
As documented. Current controller code does not need to be refactored. Controllers can continue to override the abstract function `handlePortMessage`. But if they don't, the new style is assumed.